fennel/expr/test_expr.py

```diff
@@ -0,0 +1,314 @@
+import pytest
+import pandas as pd
+from datetime import datetime
+from typing import Dict
+from fennel.datasets import dataset
+
+from fennel.expr import F, when
+from fennel.expr.visitor import ExprPrinter
+from fennel.expr.serializer import ExprSerializer
+from google.protobuf.json_format import ParseDict  # type: ignore
+from fennel.gen.expr_pb2 import Expr
+from fennel.testing.test_utils import error_message
+
+
+def test_basic_expr1():
+    expr = (F("num") + F("d")).isnull()
+    df = pd.DataFrame({"num": [1, 2, 3, 4], "d": [5, 6, 7, 8]})
+    assert expr.typeof({"num": int, "d": int}) == bool
+    ret = expr.eval(df, {"num": int, "d": int})
```

Comment:
Nit but this isn't looking very readable. I think it might be better to make `schema` a kwargs. It may end up just being about writing examples/docs where we express it like kwargs vs args without any change to the `typeof` or `eval` functions.

Reply:
yeah we should do that in the examples